FlowCrypt/Functionality/Mail Provider/Threads/MessagesThreadOperationsProvider.swift

```diff
@@ -71,7 +71,16 @@ extension GmailService: MessagesThreadOperationsProvider {
     }
 
     func archive(thread: MessageThread, in folder: String) async throws {
-        try await update(thread: thread, labelsToRemove: [.inbox, .sent])
+        // manually updated each message rather than using update(thread:...) method
+        // https://github.com/FlowCrypt/flowcrypt-ios/pull/1769#discussion_r932964129
+        try await withThrowingTaskGroup(of: Void.self) { taskGroup in
+            for message in thread.messages {
+                taskGroup.addTask {
+                    try await self.archiveMessage(message: message, folderPath: folder)
+                }
+            }
+            try await taskGroup.waitForAll()
```

Comment:
@ioanmo226 I found method for batch updating messages with Gmail API - https://developers.google.com/gmail/api/reference/rest/v1/users.messages/batchModify

Let's use it, so only 1 request will be needed for archiving any thread (even if it contains many messages)

Reply:
Sounds good!
I also worried about thread (which contains tens of messages) and this seems good solution